### ISSUE TITLE
ComboBoxItem tooltips sometimes are not displayed over their target element, but randomly misplaced

### ISSUE THREAD

**Issue by bogdan-patraucean** [microsoft/microsoft-ui-xaml#9103]
### Describe the bug

Hovering over ComboBox items that have tooltips set will display those tooltips at the wrong position sometimes. This issue started to happen since `IsConstrainedToRootBounds` was introduced in version 1.4 of WinAppSdk.

### Steps to reproduce the bug

1. Add a ComboBox
2. Place some items in it
3. Set their tooltips
4. Build the app and run it
5. Open the ComboBox
6. Hover over each item until the tooltip is misplaced

```
<ComboBox VerticalAlignment="Center" HorizontalAlignment="Center">
    <ComboBoxItem ToolTipService.ToolTip="This is a tooltip for item 1">item 1</ComboBoxItem>
    <ComboBoxItem ToolTipService.ToolTip="This is a tooltip for item 2">item 2</ComboBoxItem>
    <ComboBoxItem ToolTipService.ToolTip="This is a tooltip for item 3">item 3</ComboBoxItem>
</ComboBox>
```

### Expected behavior

I would expect that each tooltip is displayed over each ComboBoxItem.

### Screenshots

Correctly placed:
<img width="307" alt="image" src="https://github.com/microsoft/microsoft-ui-xaml/assets/26826125/75eccb65-53d8-4620-b69e-8120c8cce1f8">

Misplaced:
<img width="327" alt="image" src="https://github.com/microsoft/microsoft-ui-xaml/assets/26826125/fd31d777-2003-468c-b722-fbb1e4ac6059">

### NuGet package version

WinUI 3 - Windows App SDK 1.4.2: 1.4.231008000

### Windows version

Windows 11 (22H2): Build 22621

**Comment by ranjeshj** [comment]
Interesting bug. I'm able to repro it but very rarely. Once in like 20 tries the tooltip shows up towards the bottom right corner of the window.

**Comment by bogdan-patraucean** [comment]
Fixed with 1.5, at least from my testing it not works as expected.
